src/content/docs/en/recipes/build-forms.mdx

```diff
@@ -0,0 +1,200 @@
+---
+
+In SSR mode, Astro pages can both display and handle forms. In this recipe, you'll use a standard HTML form to submit data to the server. Your frontmatter script will handle the data on the server, sending no JavaScript to the client.
+
+## Recipe
+
+1. Create or identify a `.astro` page which will contain your form and your handling code. For example, you could add a registration page:
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    ```
+
+2. Add a `<form>` tag with some inputs to the page. Each input should have a `name` attribute that describes the value of that input. 
+
+    Be sure to include a `<button>` or `<input type="submit">` element to submit the form.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
+        <input type="text" name="username" />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" />
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+
+3. Use [validation attributes](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation) to provide basic client-side validation that works even if JavaScript is disabled. 
+
+    In this example,
+    - `required` prevents form submission until the field is filled. 
+    - `minlength` sets a minimum required length for the input text.
+    - `type="email"` also introduces validation that will only accept a valid email format.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
+        <input type="text" name="username" required />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" required />
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+    
+    :::tip
+    You can add custom validation logic, and validation logic that refers to multiple fields, using a `<script>` tag and the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#complex_constraints_using_the_constraint_validation_api).
+
+    To write complex validation logic more easily, you can build your form using a [frontend framework](/en/core-concepts/framework-components/) and choose a form library like [React Hook Form](https://react-hook-form.com/) or [Felte](https://felte.dev/).
+    :::
+
+4. Change the form's default data transfer `method` to `POST` to... or to avoid....(link this to what happens when submitting) The form submission will cause the browser to request the page again, with the form data stored in the [`Request` body](/en/reference/api-reference/#astrorequest).
```

Comment:
```suggestion
4. The form submission will cause the browser to request the page again. Change the form's data transfer `method` to `POST` to send the form data as part of the `Request` body, rather than as URL parameters.
```